plugin/trino-snowflake/src/test/java/io/trino/plugin/snowflake/TestingSnowflakeServer.java

```diff
@@ -0,0 +1,77 @@
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.snowflake;
+
+import org.intellij.lang.annotations.Language;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.Properties;
+
+import static java.util.Objects.requireNonNull;
+
+public class TestingSnowflakeServer
+        implements AutoCloseable
+{
+    public static final String TEST_URL = requireNonNull(System.getProperty("snowflake.test.server.url"), "snowflake.test.server.url is not set");
+    public static final String TEST_USER = requireNonNull(System.getProperty("snowflake.test.server.user"), "snowflake.test.server.user is not set");
+    public static final String TEST_PASSWORD = requireNonNull(System.getProperty("snowflake.test.server.password"), "snowflake.test.server.password is not set");
+    public static final String TEST_DATABASE = requireNonNull(System.getProperty("snowflake.test.server.database"), "snowflake.test.server.database is not set");
+    public static final String TEST_WAREHOUSE = requireNonNull(System.getProperty("snowflake.test.server.warehouse"), "snowflake.test.server.warehouse is not set");
+    public static final String TEST_ROLE = requireNonNull(System.getProperty("snowflake.test.server.role"), "snowflake.test.server.role is not set");
+    public static final String TEST_PROXY = System.getProperty("snowflake.test.http_proxy");
+    public static final String TEST_SCHEMA = "tpch";
+
+    public TestingSnowflakeServer()
+    {
+        execute("CREATE SCHEMA IF NOT EXISTS tpch");
+    }
+
+    public void execute(@Language("SQL") String sql)
+    {
+        execute(TEST_URL, getProperties(), sql);
+    }
+
+    private static void execute(String url, Properties properties, String sql)
+    {
+        try (Connection connection = DriverManager.getConnection(url, properties);
+                Statement statement = connection.createStatement()) {
+            statement.execute(sql);
+        }
+        catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    public Properties getProperties()
+    {
+        Properties properties = new Properties();
+        properties.setProperty("user", TEST_USER);
+        properties.setProperty("password", TEST_PASSWORD);
+        properties.setProperty("db", TEST_DATABASE);
+        properties.setProperty("schema", TEST_SCHEMA);
+        properties.setProperty("warehouse", TEST_WAREHOUSE);
+        properties.setProperty("role", TEST_ROLE);
+        return properties;
+    }
+
+    @Override
+    public void close()
+            throws Exception
+    {
+        execute("DROP SCHEMA IF EXISTS tpch");
```

Comment:
hmmm, does this mean tests cannot be executed concurrently against same snowflake instance?

Also will drop suceed without cascade?

Reply:
After upgrading trino version the Testing server switch to implement AutoCloseable. I don't know how to write close() part. So I referred to some other connectors to make it pass the compilation.